main.go

```
@@ -32,41 +33,53 @@ func main() {
 		showColors = false
 	}
 
-	fmtcolor(104, "/// g o p a c k ///")
-	fmt.Println()
 	// localize GOPATH
 	setupEnv()
 
-	deps := loadDependencies(".")
+	stats, deps := loadDependencies(".")
 
 	if os.Args[1] == "stats" {
-		p.PrintSummary()
+		stats.PrintSummary()
 	} else {
 		// run the specified command
-		runCommand()
+		runCommand(deps)
 	}
 }
 
-func loadDependencies(root string) *Dependencies {
+func announceGopack() {
+	fmtcolor(104, "/// g o p a c k ///")
+	fmt.Println()
+}
+
+func loadDependencies(root string) (*ProjectStats, *Dependencies) {
 	p, err := AnalyzeSourceTree(root)
 	if err != nil {
 		fail(err)
 	}
-	dependencies := loadConfiguration(root)
+
+	config, dependencies := loadConfiguration(root)
 	if dependencies != nil {
 		failWith(dependencies.Validate(p))
 		// prepare dependencies
 		loadTransitiveDependencies(dependencies)
+		config.WriteChecksum()
```

Yep, you're right.

I've addressed the check for exit code at 86d24ae, failing fast when the download fails.